MidtransCoreKit/MidtransCoreKit/MidtransTransactionDetails.m

```diff
@@ -26,12 +28,40 @@ - (instancetype)initWithOrderID:(NSString *)orderID andGrossAmount:(NSNumber *)g
     return self;
 }
 
+-(instancetype)initWithOrderID:(NSString *)orderID andGrossAmount:(NSNumber *)grossAmount andCurrency:(MidtransCurrency)currency {
+    if (self = [super init]) {
+        self.orderId = orderID;
+        self.grossAmount = grossAmount;
+        self.currency = currency;
+    }
+    return self;
+}
+
 - (NSDictionary *)dictionaryValue {
     // Format MUST BE compatible with
     // http://docs.veritrans.co.id/en/api/methods.html#transaction_details_attr
-    NSInteger grossAmount = [self.grossAmount integerValue];
+    NSNumber *amountNumber = nil;
+    switch ([CONFIG currency]) {
+        case MidtransCurrencyIDR:{
+            NSInteger grossAmount = [self.grossAmount integerValue];
+            amountNumber = [NSNumber numberWithInteger:grossAmount];
+        }
+            break;
+        case MidtransCurrencySGD:{
+            double grossAmount = [self.grossAmount doubleValue];
+            amountNumber = [NSNumber numberWithDouble:grossAmount];
+        }
+            break;
+        default:{
+            NSInteger grossAmount = [self.grossAmount integerValue];
+            amountNumber = [NSNumber numberWithInteger:grossAmount];
+        }
+            break;
+    }
```

Comment:
I feel this `switch ([CONFIG currency])` could be refactored to somewhere but not sure..